### PR TITLE
add token whitelist to protect against spam attacks

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: nearswap/src/deposit.rs [hunk 1]
```diff
@@ -59,6 +59,24 @@ impl NearSwap {
...
 
+    /// Record deposit of some number of tokens to this contract.
+    /// Fails if account is not registered or if token isn't whitelisted.
+    pub(crate) fn deposit_token(
+        &mut self,
+        sender_id: &AccountId,
+        token_id: &AccountId,
+        amount: Balance,
+    ) {
+        let mut account_deposit = self.get_deposit(sender_id);
```
Comment (robert-zaremba) [inline nearswap/src/deposit.rs:70]:
```suggestion
        let mut d = self.get_deposit(sender_id);
```
let's use short names

File: nearswap/src/deposit.rs [hunk 2]
```diff
@@ -153,9 +171,28 @@ pub struct AccountDeposit {
...
+    /// Remove `token_id` from this account whitelist and remove balance.
+    /// Panics if the `token_id` balance is not 0.
+    pub(crate) fn remove_from_whitelist(&mut self, token_id: &AccountId) {
+        let amount = self.tokens.remove(token_id).unwrap_or_default();
+        assert_eq!(amount, 0, "{}", ERR24_NON_ZERO_TOKEN_BALANCE);
+    }
     /**
     deposit `token`s. If this is a first depoisit, a new record is created and the minimum
     required storage is increased. */
+    /// Fails if account is not registered or if token isn't whitelisted.
```
Comment (robert-zaremba) [inline nearswap/src/deposit.rs:195]: please merge the comment with the previous one. We shouldn't mix `///` with `/** ... */`

**Review by robert-zaremba — APPROVED** [review]
Preapproving:
* unit tests are missing
* left few suggestions.

File: nearswap/src/deposit.rs (hunk 2 above)
Comment (amityadav0) [reply nearswap/src/deposit.rs:195]: ok

File: nearswap/src/deposit.rs (hunk 1 above)
Comment (amityadav0) [reply nearswap/src/deposit.rs:70]: ok

**Comment by amityadav0** [comment]
i think this cannot have unit tests because we need to make calls from owner to add or remove therefore it will have simulation tests, i will it in another PR added a task for it

**Pull request merged**